eth/interop.go

```diff
@@ -3,54 +3,58 @@ package eth
 import (
 	"context"
 	"errors"
-	"fmt"
 
-	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/core/types/interoptypes"
-	"github.com/ethereum/go-ethereum/core/vm"
-	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/miner"
+	"github.com/ethereum/go-ethereum/params"
 )
 
 func (s *Ethereum) CheckMessages(ctx context.Context, messages []interoptypes.Message, minSafety interoptypes.SafetyLevel, executingTimestamp uint64) error {
+	return errors.New("deprecated. use CheckAccessList")
+}
+
+func (s *Ethereum) CheckAccessList(ctx context.Context, inboxEntries []common.Hash, minSafety interoptypes.SafetyLevel, executingTimestamp uint64) error {
 	if s.interopRPC == nil {
-		return errors.New("cannot check interop messages, no RPC available")
+		return errors.New("cannot check interop access list, no RPC available")
 	}
-	return s.interopRPC.CheckMessages(ctx, messages, minSafety, interoptypes.ExecutingDescriptor{Timestamp: executingTimestamp})
+	return s.interopRPC.CheckAccessList(ctx, inboxEntries, minSafety, interoptypes.ExecutingDescriptor{Timestamp: executingTimestamp})
```

Comment:
See spec, for the ingress filter we want to use a 1-day timeout, set in this `ExecutingDescriptor`.
And during blockbuilding the timeout=0. It would be nice to have that flexibility, maybe the `ExecutingDescriptor` as a whole should just be a function argument here.

Reply:
Ah good callout. The structure definition in geth doesn't have Timeout in the first place, so I'll add that in, and use the Descriptor more broadly.